### PR TITLE
Fix (ci): Ensure `test-build` jobs precede `test-update` jobs to correctly test incremental builds

### PR DESCRIPTION
**Pull request by leojonathanoh**
Previously, the `test-update` job ran independently of the `test-build` job. The `test-update` job was not an incremental build based on the `test-build` job.

Now, the two independent jobs are combined as one job, with `test-build` preceeding `test-update`. This properly tests incremental builds using the local docker cache. In order to prevent `test-update` from pulling a remote `:latest` image, but to use the `:latest` from the preceeding `test-build` step's build cache, `NO_PULL` is added for `PIPELINE=update`.

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -27,6 +27,7 @@
 # LATEST=true             # bool - optional
 # CACHE=                  # bool - optional
 # NO_TEST=                # bool - optional
+# NO_PULL=                # bool - optional
```
Comment (joeltimothyoh) [inline build.sh:30]: `NO_PULL` should fall under `update` variables.

File: test/update-hlds-cstrike.env [hunk 2]
```diff
@@ -10,5 +10,6 @@ APPID=90
 GAME=cstrike
 GAME_UPDATE_COUNT=1
 NO_TEST=
+NO_PULL=true
```
Comment (joeltimothyoh) [inline test/update-hlds-cstrike.env:13]: Also `NO_PULL` should come before `NO_TEST`.

**Review by joeltimothyoh — CHANGES_REQUESTED** [review]

File: test/update-hlds-cstrike.env (hunk 2 above)
Comment (joeltimothyoh) [reply test/update-hlds-cstrike.env:13]: The PR looks generally good to me. I've fixed the positions of the switch: https://github.com/joeltimothyoh/docker-sourceservers/pull/14/commits/9069ac01c8eb434e451343c472067ecf2be4cc1e @leojonathanoh

File: build.sh (hunk 1 above)
Comment (joeltimothyoh) [reply build.sh:30]: See https://github.com/startersclan/docker-sourceservers/pull/112#discussion_r1125576877

**Comment by leojonathanoh** [comment]
your changes lgtm, please force push and merge this

**Review by joeltimothyoh — CHANGES_REQUESTED** [review]
Awaiting https://github.com/startersclan/docker-sourceservers/pull/113 before rebase

**Comment by leojonathanoh** [comment]
#113 is independent, when this is merged, it will be seen that #113 needs to fix `hlds`.

**Comment by joeltimothyoh** [comment]
ok. logs aren't going to be retained for long though. will appear like https://github.com/joeltimothyoh/docker-sourceservers/actions/runs/4335105555/jobs/7569446665

**Review by joeltimothyoh — APPROVED** [review]

**Comment by leojonathanoh** [comment]
long enough for #113 to be fixed. 😉

**Pull request merged**